### PR TITLE
Initial support for showing a caja desktop in wayland

### PR DESCRIPTION
**Pull request by lukefromdc**
*put XKillClient and gdk_x11_screen_get_xscreen behind x11-only runtime checks

### CODE REVIEW

**Comment by lukefromdc** [comment]
Out of time tonight but took a copy of the travis errors. Builds finished here fine but travis found an issue in eel-canvas.c that should predate this PR as that file was not modified.

Note that with this and it's matching mate-desktop PR, plus current panel and the applets I am working on, I now have a mate-wayfire session that is becoming increasingly difficult to tell from my x11 session!  This was supposed to be the hardest part, and I have working desktop icons in wayland, with the desktop window staying under the others and rendering on all workspaces. Interaction with it seems normal.

This will need to be checked over to make sure NO changes to x11 behavior occur and I've made as few x11 codepath changes as possible for that reason. I have to move a function though to avoid a segfault on trying to detect the monitor too soon. The desktop code is a bit racy, and that caused problems in two areas

File: libmate-desktop/mate-bg.c [hunk 1]
```diff
@@ -1163,6 +1163,7 @@ mate_bg_create_surface_scale (MateBG      *bg,
 
 	cairo_surface_t *surface;
 	cairo_t *cr;
+    GdkDisplay *display;
```
Comment (raveit65) [inline libmate-desktop/mate-bg.c:1166]: Indent

File: libmate-desktop/mate-bg.c [hunk 2]
```diff
@@ -1177,8 +1178,11 @@ mate_bg_create_surface_scale (MateBG      *bg,
 
 	mate_bg_get_pixmap_size (bg, width, height, &pm_width, &pm_height);
 
-	if (root)
+    display = gdk_display_get_default ();
```
Comment (raveit65) [inline libmate-desktop/mate-bg.c:1181]: Indent

Comment (lukefromdc) [reply libmate-desktop/mate-bg.c]: Fixed, this was one of those old files with tabs not spaces, did not show this in pluma unless looked for. Thanks

File: libmate-desktop/mate-bg.c (hunk 1 above)
Comment (raveit65) [reply libmate-desktop/mate-bg.c:1166]: Needs to be fixed....

File: libmate-desktop/mate-bg.c [hunk 3]
```diff
@@ -1177,8 +1178,11 @@ mate_bg_create_surface_scale (MateBG      *bg,
 
 	mate_bg_get_pixmap_size (bg, width, height, &pm_width, &pm_height);
 
-	if (root)
+	display = gdk_display_get_default ();
+
+	if ((root) &&  GDK_IS_X11_DISPLAY (display))
 	{
+
 		surface = make_root_pixmap (window, pm_width * scale, pm_height * scale);
```
Comment (raveit65) [inline libmate-desktop/mate-bg.c:1186]: unneeded empty line 1185.

File: libmate-desktop/mate-bg.c (hunk 1 above)
Comment (raveit65) [reply libmate-desktop/mate-bg.c:1166]: Btw. here a tab is needed.

File: libmate-desktop/mate-bg.c (hunk 3 above)
Comment (lukefromdc) [reply libmate-desktop/mate-bg.c:1185]: unneded line removed

File: libmate-desktop/mate-bg.c (hunk 1 above)
Comment (lukefromdc) [reply libmate-desktop/mate-bg.c]: Found and fixed, BTW that Pluma setting to show tabs and spaces is a real game changer for code!

**Review by raveit65 — APPROVED** [review]
LGTM

**Pull request merged**